caluma/caluma_workflow/serializers.py

```diff
@@ -229,56 +171,24 @@ class CaseSerializer(SendEventSerializerMixin, serializers.ModelSerializer):
     )
 
     def validate(self, data):
-        form = data.get("form")
-        workflow = data["workflow"]
-
-        if form:
-            if (
-                not workflow.allow_all_forms
-                and not workflow.allow_forms.filter(pk=form.pk).exists()
-            ):
-                raise exceptions.ValidationError(
-                    f"Workflow {workflow.pk} does not allow to start case with form {form.pk}"
-                )
+        try:
+            data = models.Case.objects._validate(data)
+        except Exception as e:
+            raise exceptions.ValidationError(str(e))
 
         return super().validate(data)
 
     @transaction.atomic
     def create(self, validated_data):
         user = self.context["request"].user
-        parent_work_item = validated_data.get("parent_work_item")
-        validated_data["status"] = models.Case.STATUS_RUNNING
-
-        form = validated_data.pop("form", None)
-        if form:
-            validated_data["document"] = Document.objects.create(
-                form=form, created_by_user=user.username, created_by_group=user.group
-            )
 
-        if parent_work_item:
-            case = parent_work_item.case
-            while hasattr(case, "parent_work_item"):
-                case = case.parent_work_item.case
-            validated_data["family"] = case
+        validated_data = models.Case.objects._pre_create(validated_data, user)
```

Comment:
I don't like this 3-step creation procedure. Users *may* forget to do it exactly this way (granted, much better than before, but still)

Also, underscore-prefixed names in python denote private methods, so this looks very much like a violation to me. Especially since we want to simplify the procedure for people using our code directly, this is a bad idea IMHO